v2/blockstore/readwrite.go

```diff
@@ -178,6 +179,12 @@ func (b *ReadWrite) initWithRoots(v2 bool, roots []cid.Cid) error {
 	return carv1.WriteHeader(&carv1.CarHeader{Roots: roots, Version: 1}, b.dataWriter)
 }
 
+// Index gives direct access to the index.
```

Comment:
do we need this on `ReadWrite` - that's where it's dangerous.

The worry with `ReadWrite` access is you add another record, and then when you go to finalize the blockstore the extra record creeps in.

on ReadOnly, the index you access isn't going to get written back out into the underlying car file.

Reply:
I do need that on `ReadWrite`, but I can add a wrapper to enforce the read-only access.

Reply:
do you need it before you finalize?
could we have [finalizeReadOnly](https://github.com/ipld/go-car/blob/master/v2/blockstore/readwrite.go#L257) give you a `ReadOnly` variant that you can then get the index from?

Reply:
I only need it after finalize, but I sort of see why some might.